lib/fluent/supervisor.rb

```diff
@@ -806,6 +811,53 @@ def configure(supervisor: false)
 
       $log.info :supervisor, 'parsing config file is succeeded', path: @config_path
 
+      # merge additional configuration
+      config_type = if File.extname(@config_path) == '.yaml' || File.extname(@config_path) == '.yml'
+                      :yaml
+                    else
+                      nil
+                    end
+      if @system_config.config_include_dir&.empty?
+        $log.warn :supervisor, 'default configuration include directory was disabled'
+      else
+        begin
+          Find.find(@system_config.config_include_dir) do |path|
+            next if path == @system_config.config_include_dir
```

Comment:
```suggestion
            next unless File.file?(path)
```

`Find.find` search entries recursively including sub-directories, right?
If so, we need to skip all directories.